website/blog/2024-03-18-deep-learning-part-2/index.md

```diff
@@ -0,0 +1,230 @@
+
+
+# Use Case: Fine Tuning Llama2-7B for NER on Restaurant data
+
+Let's consider fine-tuning a Llama2-7B model for following instructions embedded within text. This is a specific example of how Parameter-Efficient Fine-Tuning (PEFT) can be applied for real-world tasks. The goal here is to  extract structured information based on instructions, which goes beyond simple Named Entity Recognition (NER).
+
+Imagine a scenario where the instruction is: `The Punter is an expensive Chinese coffee shop located near Café Sicilia.` The model, after fine-tuning, would then process the unstructured text of the sentence and identify the entities, presenting them in a structured format like a list. 
+
+Example of extracting information from `The Punter is an expensive Chinese coffee shop located near Café Sicilia.`: 
+```python
+Business name: The Punter
+Cusine: Chinese
+descriptor: expensive
+location: Café Sicilia
+```
+
+The Llama2-7B model's size is strategically chosen. It has enough parameters (complexity) to handle intricate tasks like understanding instructions within text, but remains  compact enough for fine-tuning on even a modest cluster of less powerful GPUs like Nvidia T4s, thanks to PEFT methods. It can even be fine-tuned on a single high-end A100 GPU with PEFT.
+
+PEFT techniques focus on keeping the pre-trained model parameters frozen and only training a small number of additional layers for the specific task. This saves memory and training time.
+
+Overall, fine-tuning Llama2-7B with PEFT is a promising approach for instruction following tasks and similar applications that require working with large models on limited resources.
+
+
+
+# Setting Up MLflow for Fine tuning
+
+The code provided below illustrates the setup for Fine tuning, augmented with MLflow integration. We have marked the sections used for MLflow tracking. At the beginning of the functions, MLflow is configured, initiating a run that automatically logs the experiment. The details of  `callback.py` will be discussed in a later section.
+
+This example demonstrates how to set up a fine tuning job on Databricks. A similar setup can be achieved on any platform capable of supporting this type of workload.
+
+We are using the [e2e_nlg](https://huggingface.co/datasets/e2e_nlg) dataset as it is a great dataset is used for training end-to-end, data-driven natural language generation systems in the restaurant domain
+
+```bash
```

Comment:
Do we need to present this in this way? Is the intention to show readers how to execute a script within a notebook, or is it to show how to build the model training logic that integrates with MLflow? 
I'm not convinced that this is useful to include.